src/audio/module_adapter/module_adapter.c

```diff
@@ -1291,8 +1319,20 @@ int module_adapter_trigger(struct comp_dev *dev, int cmd)
 		dev->state = COMP_STATE_ACTIVE;
 		return PPL_STATUS_PATH_STOP;
 	}
-	if (interface->trigger)
+
+	if (interface->trigger) {
+#if CONFIG_IPC_MAJOR_4
+		if (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP) {
+			/* Process DP module's trigger */
+			union scheduler_dp_thread_ipc_param param = {
+				.pipeline_state.trigger_cmd = cmd,
+			};
+			return scheduler_dp_thread_ipc(mod, SOF_IPC4_GLB_SET_PIPELINE_STATE,
+						       &param);
+		}
+#endif
```

Comment:
Please wrap this in an #ifdef, for instance `CONFIG_SOF_USERSPACE_ALL_DP_MODS` or `CONFIG_SOF_DP_THREAD_HANDLE_IPC`.